From a07fb96b9181be4e5f827e8a5128279e9efdc892 Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@marvell.com>
Date: Fri, 20 Nov 2020 10:56:52 +0530
Subject: [PATCH 0941/1921] octeontx2-pf: Fix unmapping LMTST region

Unmap LMTST region only after stoping TX/RX queues.

Change-Id: I98a11226b854bc067a565fdc5180559b2928a603
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/40529
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 7 ++-----
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c | 4 +---
 2 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index 8e460f7c4374..b9485839cfd5 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -2875,10 +2875,6 @@ static void otx2_remove(struct pci_dev *pdev)
 		return;
 
 	pf = netdev_priv(netdev);
-
-	if (pf->hw.lmt_base)
-		iounmap(pf->hw.lmt_base);
-
 	pf->flags |= OTX2_FLAG_PF_SHUTDOWN;
 
 	if (pf->flags & OTX2_FLAG_TX_TSTAMP_ENABLED)
@@ -2895,8 +2891,9 @@ static void otx2_remove(struct pci_dev *pdev)
 	otx2_sriov_disable(pf->pdev);
 	otx2_ptp_destroy(pf);
 	otx2_mcam_flow_del(pf);
-
 	otx2_detach_resources(&pf->mbox);
+	if (pf->hw.lmt_base)
+		iounmap(pf->hw.lmt_base);
 	otx2_disable_mbox_intr(pf);
 	otx2_pfaf_mbox_destroy(pf);
 	pci_free_irq_vectors(pf->pdev);
diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
index 9242f216d5fa..b4b618277096 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
@@ -699,11 +699,9 @@ static void otx2vf_remove(struct pci_dev *pdev)
 		unregister_netdev(netdev);
 
 	otx2vf_disable_mbox_intr(vf);
-
+	otx2_detach_resources(&vf->mbox);
 	if (vf->hw.lmt_base)
 		iounmap(vf->hw.lmt_base);
-
-	otx2_detach_resources(&vf->mbox);
 	otx2vf_vfaf_mbox_destroy(vf);
 	pci_free_irq_vectors(vf->pdev);
 	pci_set_drvdata(pdev, NULL);
-- 
2.31.1

